resources/lambdas.md

```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
 This will be a basic introduction into functional programming, I recommend first reading the earlier section on lambdas and our section on [streams](/resources/streams) as it will help a lot and since 
 most of functional programming is streams and lambdas.
 
-## What is Functional Programming
+## What is a Functional Programming Language?
 
-Functional programming is a programming paradigm where functions are "treated as first class citizens". Essentially, functions are treated as objects and can do
-anything that an object can (be passed as a parameter, returned, etc).
-The main idea behind functional programming is that it's programming without any or with minimal [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). 
-In other words, it's programming where only local variables that are created inside the function can be modified. This means that no global variables, variables passed by reference, or parameterized variables can be
-modified for it to count as functional programming. An example would be:
+In general programming theory, there exists two main types of languages:
+
+- [Imperative programming](https://en.wikipedia.org/wiki/Imperative_programming)
+- [Functional programming](https://en.wikipedia.org/wiki/Functional_programming)
+
+While these two will turn out to seem super different, they are [proven](https://en.wikipedia.org/wiki/Church%E2%80%93Turing_thesis) to be identical.
+
+In a nutshell, functional programming languages treat EVERYTHING as a function, including the basic definitions of `true` and `false`!
+
+From there, every series of actions is not defined as **what something should do**, but rather **what something is** as a recursive set of functions, then asking for a thing.
```

Comment:
These few lines of explanations above are not very intuitive.

Reading this line specifically, I'm still not entirely sure what it means. Maybe reword for clarity?

Reply:
Reworded in terms of "problem solving" and "languages" to show how they interact.
Hopefully it makes it more clear.